pkg/controller/namespace/deletion/namespaced_resources_deleter.go

```diff
@@ -471,6 +471,13 @@ func (d *namespacedResourcesDeleter) deleteAllContentForGroupVersionResource(
 				return finalizerEstimateSeconds, nil
 			}
 		}
+
+		if gvr.GroupResource() == (schema.GroupResource{Group: "", Resource: "pods"}) {
```

Comment:
Shouldn't d.estimateGracefulTermination be the one calculating this?  Why are we checking a second time?

I would have expected `d.estimateGracefulTermination` to return a non-zero

Reply:
I am not sure I understand your comment here. It is preexisting that finalizers are estimated in this method just a few lines above (https://github.com/kubernetes/kubernetes/pull/80479/files#diff-e26e1ddd43828aa03378907fb36566aaR471). This is just adding a finalizer that isn't explicitly stated on the Pod object but is implied.

`estimateGracefulTermination` is purely about reading the `TerminationGracePeriodSeconds` value from object which seems distinct from finalizers, at very least preexisting.